src/com/googlecode/jmxtrans/model/output/GraphiteWriter.java

```diff
@@ -101,6 +101,11 @@ public void internalWrite(Server server, Query query, ImmutableList<Result> resu
 							log.debug("Graphite Message: {}", line);
 							writer.write(line);
 							writer.flush();
+							if (writer.checkError()) {
+								log.error("Error writing to Graphite, clearing Graphite socket pool");
+								pool.returnObject(address, socket);
+								pool.clear();
```

Comment:
Clearing the pool to get rid of a single dirty connection is a bit overkill. It is also non thread safe as there is a chance that the dirty connection is out of the pool again when we clear it (and `clear()` only clears the idle objects. Using `invalidateObject()` is probably better.

That being said, good catch ! We've been having the same kind of problems, but could not pinpoint the cause !